### PR TITLE
travis: remove go1.12 job

### PR DESCRIPTION
**Pull request by skylenet**
This is not needed. The Ubuntu PPA is using go1.11.

### CODE REVIEW

File: .travis.yml
Comment (janos) [inline .travis.yml:24]: Go 1.12 is not needed for PPA, but it is a supported Go version. Last two releases are. Maybe we should keep it for this reason and change the comment?

Comment (skylenet) [reply .travis.yml:24]: Why do we need to support go1.12?

Comment (janos) [reply .travis.yml:24]: I thought that we supported current and previous go version. I could be wrong. If we are supporting only the current, then the job is not needed.

Comment (skylenet) [reply .travis.yml:24]: Previously we were supporting 1.11 and 1.12 mainly because the Ubuntu PPA needs 1.11. 

IMO we can remove 1.12 because none of our build systems need it. Our binaries are published using 1.13.

I'm also in favor of getting rid of 1.11 and the Ubuntu PPA builds, but that's a topic for the roundtable :)

Comment (janos) [reply .travis.yml:24]: Maybe our codebase does not need to ensure support for previous go version for people that cannot upgrade right away, as it is easy to have two version one beside another. In that case, removing this job is ok.

Comment (nonsense) [reply .travis.yml:24]: It is up to us to decide what to support. One less version means faster CI, but might be a problem who haven't upgraded to Go 1.13.

I don't have a strong feeling, but I like faster CI, so I am also up for removing Go 1.12.

Comment (janos) [reply .travis.yml:24]: I do not say that anybody else should decide on what we should support, this is not what I meant. I am sorry if that is the impression that I made. I just wrongly assumed that we care about the people who are more conservative when upgrading Go versions.

I believe that these CI jobs are running in parallel, so I am not sure about the gain in that area.

Anyway, I will not block this change. It looks like that there is no need for us to support current-1 Go version.

**Review by janos — APPROVED** [review]

File: .travis.yml
Comment (zelig) [reply .travis.yml:24]: i also thought it was deliberate check for backward compatibility one version back. I would leave it. is it really making CI faster given parallelism?

Comment (skylenet) [reply .travis.yml:24]: Backward compatibility is not needed IMO. Our binaries are released with go1.13, so why should we care about supporting go1.12?

Comment (janos) [reply .travis.yml:24]: Given that this is an open source project, someone may still use older but still supported Go version to build or use some of the packages from swarm. Maybe we still did not reach that level of project maturity or popularity, so ensuring that project works only with the latest release is just fine, and that we can think about this later.

Comment (nonsense) [reply .travis.yml:24]: Just a comment on the CI from my side - we have 5 concurrent jobs in total for the `ethersphere` organisation, so if half of them are used on jobs we don't need, CI is essentially twice slower, specially when you have to wait for free executors - yesterday we waited for one-two hours to get helm charts released, due to many PRs being submitted at the same time.

Comment (nonsense) [reply .travis.yml:24]: `Maybe we still did not reach that level of project maturity or popularity, so ensuring that project works only with the latest release is just fine, and that we can think about this later.` - exactly my opinion on this PR.

**Review by nonsense — APPROVED** [review]

**Comment by skylenet** [comment]
Thanks for the reviews and valid opinions. I'll merge it for now. We can still add it back If we ever have a case where someone need this to be compatible with another go version.

**Pull request merged**
